core/workloads/containerized_workload/containerized_workload.md

```diff
@@ -0,0 +1,288 @@
+# Containerized Workload
+A containerized workload is used for long-running workloads in containers. Here is the definition of the containerized workload.
```

Comment:
This is where execution policies are needed. A containerized workload need not strictly be a long-running workload.

Reply:
IMO, there are two ways to solve this.

- One is to rename this to be `containerized server` and add another workload called `containerized task` which has a different schema (task specific) from server. 

- The other is to keep this as the only core containerized workload and add policies for components.

These two ways can actually co-exist too.

Reply:
> which has a different schema (task specific) from server

How do we expect task specific schema would differ? Under the contemporary spec the schema of task and server (and worker) are identical.

Reply:
I don't think they would differ at all. That's the problem we solved with workloadType in OAM alpha1 and have now re-introduced here. Rather than define multiple workload definitions with identical schemas, I think we need a way to classify execution behavior or something like that, so that we avoid duplicate schema definitions and have a way to express whether or not a trait is applicable to a given component.

Reply:
@vturecek To be honest it seems pretty reasonable to me for workloads that are conceptually different things to encode that difference in their GVK, even if they have identical schema. Previously we effectively had a `ComponentSchematic` type with a subtype encoded in `spec.workloadType`. Now we have a `Component` type with a subtype encoded in `spec.workload.kind`.

I could see a case for moving the encoding of whether or not something was a singleton into the schema as a field under `spec` and simply having three kinds with identical schemas; probably `ContainerizedServer`, `ContainerizedWorker`, and `ContainerizedTask` similar to what @ryanzhang-oss suggested.

Reply:
> simply having three kinds with identical schemas; probably `ContainerizedServer`, `ContainerizedWorker`, and `ContainerizedTask` similar to what @ryanzhang-oss suggested.

Since those are all core workloads, we need to spell out their schematic in the spec. It seems we either 
- have 3 identical schemas in the spec 
- explain that we use the schema's name to differentiate the runtime characteristics of a workload from another workload with an identical schema.

I think the second approach is pretty much the same as in spec alphav1 as @negz says. I am just not sure how can we evolve it to the policy approach which I think we all agree is the right direction.

Let's find a time to discuss this in a meeting.